crates/namada/src/vm/wasm/compilation_cache/common.rs

```diff
@@ -36,6 +36,12 @@ pub struct Cache<N: CacheName, A: WasmCacheAccess> {
     name: PhantomData<N>,
     /// Cache access level
     access: PhantomData<A>,
+    /// Wasmer store - The store that's used to compile the modules has to be
+    /// kept alive to avoid segfaults when running them from cache (otherwise
+    /// `wasmer_compiler::FRAME_INFO` gets cleared out when the `Store` is
+    /// dropped and the next run of this `Module` doesn't re-instantiate it and
+    /// crashes when it tries to access it)
+    store: Arc<Store>,
```

Comment:
According to https://github.com/wasmerio/wasmer/issues/4949#issuecomment-2245453917, modules from different stores should not be interchanged. I'm not sure what the consequences are... but we should watch out for weird behavior stemming from this.

Reply:
also a minor nit is, the majority of this text is comprised of details that don't belong in a docstring IMO. part of it can be sliced out into a regular comment prefixed by `NB`, `NOTE`, or something along these lines

Reply:
yeah, I think we have to live with it until it's fixed upstream as:

- using the same store everywhere leads to a leak
- using a new store without keeping this one in cache leads to invalid memory access segfault

From what I can see the consequence is that the store in cache will never be cleared, but because we only allow a limited subset of wasms to be compiled and cached we should be fine for now. In dry-run txs the read-only cache access doesn't use this store from cache so it should not be possible to grow it from it.

Reply:
ah, actually, read-only cache access does use this `store` - let me fix that

Reply:
added
- [32b2682](https://github.com/anoma/namada/pull/3529/commits/32b268211c5e2048bcfcde775fa6884cd0c9a7c9) to fix the read-only cache to use a temp store
- [163d393](https://github.com/anoma/namada/pull/3529/commits/163d393266b336726e4595e164b81b876d15cd22) to fix the comment